src/components/Analysis/Purposes/PurposesList.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+
+import { isQueryLoading, useQuery } from 'cozy-client'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+
+import { buildGeoJSONQueryNoLimit } from 'src/queries/queries'
+import LoadedPurposesList from 'src/components/Analysis/Purposes/LoadedPurposesList'
+
+const PurposesList = () => {
+  const geoJsonQuery = buildGeoJSONQueryNoLimit()
+  const { data: timeseries, ...queryResult } = useQuery(
+    geoJsonQuery.definition,
+    geoJsonQuery.options
+  )
+
+  const isLoading = isQueryLoading(queryResult)
+
+  if (isLoading) {
+    return (
+      <Spinner size="xxlarge" className="u-flex u-flex-justify-center u-mt-1" />
+    )
+  }
+
+  return <LoadedPurposesList timeseries={timeseries} />
```

Comment:
personnally I would have tested the 2 states of the component (loading = true and false)

and i would have checked useQuery and isQueryLoading are called with correct params.

if you want any help, feel free to ping me

Reply:
```
import React from 'react'
import { render } from '@testing-library/react'
import PurposesList from './PurposesList'
import { isQueryLoading, useQuery } from 'cozy-client'

import { buildGeoJSONQueryNoLimit } from 'src/queries/queries'

jest.mock('cozy-client', () => ({
  isQueryLoading: jest.fn(),
  useQuery: jest.fn()
}))
jest.mock('cozy-ui/transpiled/react/Spinner', () => ({ size, className }) => (
  <div data-testid="Spinner" className={className} data-size={size} />
))
jest.mock('src/queries/queries')
jest.mock(
  'src/components/Analysis/Purposes/LoadedPurposesList',
  () => ({ timeseries }) => (
    <div data-testid="LoadedPurposesList" data-timeseries={timeseries} />
  )
)

describe('PurposesList', () => {
  beforeEach(() => {
    buildGeoJSONQueryNoLimit.mockReturnValue({
      definition: 'definition',
      options: 'options'
    })
    useQuery.mockReturnValue({
      data: ['timeseries'],
      other: 'value'
    })
  })

  it('should call use query with correct definition from build geo json', () => {
    // When
    render(<PurposesList />)

    // Then
    expect(useQuery).toHaveBeenCalledWith('definition', 'options')
  })

  it('should display large spinner when is query loading', () => {
    // Given
    isQueryLoading.mockReturnValue(true)

    // When
    const { getByTestId } = render(<PurposesList />)

    // Then
    expect(getByTestId('Spinner').getAttribute('data-size')).toEqual('xxlarge')
  })

  it('should display loaded purposes list when is not query loading', () => {
    // Given
    isQueryLoading.mockReturnValue(false)

    // When
    const { getByTestId } = render(<PurposesList />)

    // Then
    expect(
      getByTestId('LoadedPurposesList').getAttribute('data-timeseries')
    ).toEqual('timeseries')
  })
})


```

Reply:
et en plus j'ai oublié :
```
  it('should detect if query loading is true when query result', () => {
    // When
    render(<PurposesList />)

    // Then
    expect(isQueryLoading).toHaveBeenCalledWith({ other: 'value' })
  })
  ```

Reply:
merci c'est plus clair ! Mes points de vue : 
- le 1 pour moi c'est de l'implem technique et il peut y en avoir plusieurs de différentes. C'est pas un pré-requis indispensable au composant
- le 2 idem, de plus le texte du should n'est pas tout à fait juste par rapport au test (ou alors j'ai pas compris)
- je suis pas pour mocker useQuery en lui faisant retourner des valeurs qui ne sont pas celle de l'API
- le 3 et 4 ok pour ça 👍 , même si mocker le result de la useQuery me paraît plus juste (que de mocker le isQueryLoading)